luxonis_ml/data/README.md

```diff
@@ -0,0 +1,611 @@
+    # the polyline will be closed to form a polygon,
+    #   i.e. the first and last point are the same
+    "polyline": list[tuple[float, float]],
+}
+```
+
+#### Binary Mask
+
+The mask is a binary 2D numpy array.
+
+```python
+{
+    # type of the annotation, always "mask"
+    "type": "mask",
+
+    # name of the class this mask belongs to
+    "class": str,
+
+    # binary mask as a 2D numpy array
+    # 0 for background, 1 for the object
+    "mask": np.ndarray,
+}
+```
+
+#### Run-Length Encoding
+
+The mask is described using the [Run-Length Encoding](https://en.wikipedia.org/wiki/Run-length_encoding) compression.
+
+Run-length encoding compresses data by reducing the physical size
+of a repeating string of characters.
+This process involves converting the input data into a compressed format
+by identifying and counting consecutive occurrences of each character.
+
+The RLE is composed of the height and width of the mask image and the counts of the pixels belonging to the positive class.
+
+```python
+{
+    # type of the annotation, always "rle"
+    "type": "rle",
+
+    # name of the class this mask belongs to
+    "class": str,
+
+    # height of the mask
+    "height": int,
+
+    # width of the mask
+    "width": int,
+
+    # counts of the pixels belonging to the positive class
+    "counts": list[int] | bytes,
+
+}
+```
+
+{% alert %}
+The RLE format is not intended for regular use and is provided mainly to support datasets that may already be in this format.
+{% /alert %}
+
+{% alert %}
+Masks provided as numpy arrays are converted to RLE format internally.
+{% /alert %}
+
+### Array
+
+An array of arbitrary data. This can be used for any custom data that doesn't fit into the other types.
+
+```python
+{
+    # type of the annotation, always "array"
+    "type": "array",
+
+    # name of the class this array belongs to
+    "class": str,
+
+    # path to a `.npy` file containing the array data
+    "path": str,
+}
+```
+
+## Augmentation
+
+`LuxonisLoader` provides a way to use augmentations with your datasets. Augmentations are transformations applied to the data to increase the diversity of the dataset and improve the performance of the model.
+
+### Augmentation Pipeline
+
+The augmentations are specified as a list of dictionaries, where each dictionary represents a single augmentation. The structure of the dictionary is as follows:
+
+```python
+{
+    "name": str,  # name of the augmentation
+    "params": dict  # parameters of the augmentation
+}
+```
+
+By default, we support most augmentations from the `albumentations` library. You can find the full list of augmentations and their parameters in the {% link href="https://albumentations.ai/docs/api_reference/augmentations/" %}Albumentations documentation{% /link %}.
+
+On top of that, we provide a handful of custom batch augmentations:
+
+- `Mosaic4` - Mosaic augmentation with 4 images. Combines crops of 4 images into a single image in a mosaic pattern.
```

Comment:
Should we include params of these custom augs we provide? Or maybe even better, have a separate README in the `data/augmentations/custom` where all custom augs are documented and link to that from here?

Reply:
Yeah it would make sense to add a separate README for the augmentation that would include more information. I think we can add this as a part of the augmentations refactor?

Reply:
Yes, that makes sense